irohad/ametsuchi/storage.hpp

```diff
@@ -70,6 +60,13 @@ namespace iroha {
       virtual expected::Result<void, std::string> insertPeer(
           const shared_model::interface::Peer &peer) = 0;
 
+      /**
+       * Creates a mutable storage from the current state
+       * @return Created Result with mutable storage or error string
+       */
+      virtual expected::Result<std::unique_ptr<MutableStorage>, std::string>
+      createMutableStorage(BlockStorageFactory &storage_factory) = 0;
```

Comment:
Clang complains that this function hides overloaded virtual function from class `MutableFactory`. Maybe we should better to move this declaration there too?